pandaserver/daemons/utils.py

```diff
@@ -166,10 +167,21 @@ def got_end_sig(sig, frame):
             if to_run_daemon:
                 last_run_start_ts = int(time.time())
                 try:
-                    # execute the module script with arguments
-                    tmp_log.info('{dem} start'.format(dem=dem_name))
-                    the_module.main(argv=mod_argv, tbuf=tbuf)
-                    tmp_log.info('{dem} finish'.format(dem=dem_name))
+                    if is_loop:
+                        # go looping the script until reaching daemon period
+                        tmp_log.info('{dem} start looping'.format(dem=dem_name))
+                        start_ts = time.time()
+                        while True:
+                            the_module.main(argv=mod_argv, tbuf=tbuf)
+                            now_ts = time.time()
+                            if now_ts > start_ts + dem_period:
```

Comment:
It will go to the next loop even if 

jor_list = taskBuffer.listJobOutputReport(only_unlocked=True, time_limit=10, limit=n_jors_per_batch*nThr

in add_main gives an empty list. Generally you can skip the next loops when
len(jor_list) < n_jors_per_batch*nThr. How about adding an if-else based on
a return code from main()?

Reply:
Let me try